### PR TITLE
Docs: Minimally modernize Sphinx configuration. Fix building on Python 3.11.

### PR DESCRIPTION
**Pull request by amotl**
## About
Just a little maintenance patch for the Sphinx docs, to minimally modernize dependencies, and to fix the build [^1].

[^1]: [...] and to probe if any commit styles of mine (commit messages, wording, whatever) need to be adjusted to comply with any policies employed here.

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -83,7 +83,7 @@ Testimonials
 
 
 .. _Django REST Framework: https://www.django-rest-framework.org/
-.. _Two Scoops of Django: https://www.twoscoopspress.com/products/two-scoops-of-django-1-11
+.. _Two Scoops of Django: https://www.feldroy.com/two-scoops-press#two-scoops-of-django
```
Comment (amotl) [inline docs/source/index.rst:86]: Let's add a little workflow invoking `make linkcheck`?

**Comment by kennethreitz** [comment]
:sparkles: :cake: :sparkles:

File: docs/source/index.rst (hunk 1 above)
Comment (amotl) [reply docs/source/index.rst:86]: ... too late, merged already. ;]

**Comment by kennethreitz** [comment]
oh sorry!

**Comment by kennethreitz** [comment]
Sorry, it's been a while since I've done this :)

File: docs/source/index.rst (hunk 1 above)
Comment (kennethreitz) [reply docs/source/index.rst:86]: @amotl that's a good idea

**Pull request merged**